lightly/api/api_workflow_client.py

```diff
@@ -119,14 +122,17 @@ def _order_list_by_filenames(self, filenames_for_list: List[str], list_to_order:
                 Some values belonging to the samples
 
         Returns:
-            The list reordered. The same reorder applied on the filenames_for_list
-            would put them in the order of the filenames in self.filenames_on_server
+            The list reordered.
+            The same reorder applied on the filenames_for_list would put them
+            in the order of the filenames in self.filenames_on_server.
+            every filename in self.filenames_on_server must be in the
+            filenames_for_list.
 
         """
         assert len(filenames_for_list) == len(list_to_order)
+        assert len(filenames_for_list) == len(self.filenames_on_server)
```

Comment:
I must have missed this the last time but don't use asserts here, rather raise an exception.

Reply:
I replaced with a ValueError and wrote a unittest for it.